src/views/Profile/PublicProfile.tsx

```diff
@@ -80,12 +82,21 @@ const Section = styled.div`
 const PublicProfile = () => {
   const { account } = useWeb3React()
   const { profile } = useProfile()
+  const [usernameVisibilityToggled, setUsernameVisibility] = useState(false)
```

Comment:
Use the `usePersistState` hook so the setting is saved in local storage.

Reply:
With this library https://github.com/pancakeswap/pancake-frontend/pull/943/commits/068a828bb10ec8659d8977694177eeb243cb9473?

Reply:
Sorry, I should have been more explicit.
https://github.com/pancakeswap/pancake-frontend/blob/develop/src/hooks/usePersistState.ts

Reply:
Ah nice, last time I've rebased upstream it wasn't there, hence why I've added a new dependency.

Done with https://github.com/pancakeswap/pancake-frontend/pull/943/commits/de38ae8d052a6038850c4e81c64dd624298d7048